geps/gep-2198.md

```diff
@@ -0,0 +1,105 @@
+# JWT Policy
+
+## TLDR
+
+This document is not intended to describe how a proxy implements JWT handling behavior, there is an RFC that addresses that.
+Rather, this document is to form a common understanding or JWT use cases as it relates to configuring and handling JWTs as part of a workflow as it relates to a Gateway and establish a framework from which to build the common parts of the Policy.
+
+## Goals
+
+- Define how JWT is used to form a common understanding and establish a common language
+- Identify unique use cases for JWT with proxies
+- Propose some properties to form the basis for discussion
+
+## Non-Goals
+
+- Define how proxies implement the JWT standards
+
+## What is a JWT
+
+A [JSON web token](https://www.rfc-editor.org/rfc/rfc7519) is a common pattern of presenting authorization data (claims) between a front end or client system and a back end.
+A JWT by itself it is only one piece of the [authentication triad](https://csrc.nist.gov/glossary/term/Multi_Factor_Authentication#:~:text=The%20three%20authentication%20factors), which is "something you have".  We will assume that some other process satisfied "something you are" and "something you know".  And thus the JWT is generally considered the result of some type of authentication or formation of a trust.
+
+JWT's can be encrypted, have methods for validation, can contain a variety of claims, and be used in gate keeping (validation) and authorization (claims matching) workflows.
+
+## How is a JWT used
+
+In practice a JWT is presented to an authorization layer to make a decision of whether or not to grant access to a specific hostname or application path (backend service) or also possibly used by the backend service to decide if specific secured data can be presented.
+This bring forth a set of common workflows which in turn present the use cases. In this example we will walk through the life of an encrypted JWT (though not all JWTs are encrypted)
+
+- client obtains a JWT
+- client presents the JWT with the request to a Gateway
+- the Gateway decrypts the JWT and validates it
+- now that the JWT is proven valid it is implicitly trusted and the request processing continues
+
+At this point a number of and/or paths can take place, most commonly a combination of the following happens:
+
+- the token content is tested for the existence of some claim property or value
+- claim values may be extracted and saved as header value(s) to the request
+- the entire token or a sub-set of claims may be forwarded directly to backend services for evaluation
```

Comment:
Important use: a different token signed by the gateway is forwarded to backends ( to prevent backends from accessing the original JWT which may have escalation risks). That's sometimes called 'mint'